pkg/ext-proc/backend/llmserverpool_reconciler.go

```diff
@@ -21,39 +36,167 @@ const (
 // will have the proper controller that will create/manage objects on behalf of the server pool.
 type LLMServerPoolReconciler struct {
 	client.Client
-	Scheme         *runtime.Scheme
-	Record         record.EventRecorder
-	ServerPoolName string
-	Namespace      string
-	Datastore      *K8sDatastore
-	Zone           string
+	//Scheme         *runtime.Scheme
+	Record            record.EventRecorder
+	ServerPoolName    string
+	Namespace         string
+	Datastore         *K8sDatastore
+	Zone              string
+	v1alpha1clientset clientset.Interface
+	poolLister        listers.LLMServerPoolLister
+	poolSynced        cache.InformerSynced
+
+	workqueue workqueue.TypedRateLimitingInterface[cache.ObjectName]
+	// Still uses recorder
+}
+
+func NewLLMServerPoolReconciler(
+	//Params:
+	ctx context.Context,
+	serverPoolInformer informers.LLMServerPoolInformer,
+	clientset clientset.Interface,
+	kubeclientset kubernetes.Interface,
+	scheme *runtime.Scheme,
+	serverpoolName string,
+	namespace string,
+	zone string,
+	datastore *K8sDatastore,
+	//Return val:
+) *LLMServerPoolReconciler {
+	// Body:
+	//utilruntime.Must(v1alpha1scheme.AddToScheme(scheme.Scheme))
+
+	klog.V(4).Info("Creating event broadcaster")
+	eventBroadcaster := record.NewBroadcaster(record.WithContext(ctx))
+	eventBroadcaster.StartStructuredLogging(0)
+	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{Interface: kubeclientset.CoreV1().Events("")})
+	recorder := eventBroadcaster.NewRecorder(scheme, corev1.EventSource{Component: "FIGUREOUT NAMING HERE"})
+	ratelimiter := workqueue.NewTypedMaxOfRateLimiter(
+		workqueue.NewTypedItemExponentialFailureRateLimiter[cache.ObjectName](5*time.Millisecond, 1000*time.Second),
+		&workqueue.TypedBucketRateLimiter[cache.ObjectName]{Limiter: rate.NewLimiter(rate.Limit(50), 300)},
+	)
+
```

Comment:
it seems all the setup above: creating even recorder and the workqueue can be placed in a common pkg that we can reuse across reconcilers